meilisearch-schema/src/schema.rs

```diff
@@ -118,8 +110,10 @@ impl Schema {
                 Ok(id)
             }
             None => {
-                self.set_indexed(name)?;
-                self.set_displayed(name)
+                let id = self.fields_map.insert(name)?;
+                let pos = self.indexed_map.len() as u16;
+                self.indexed_map.insert(id, pos.into());
+                Ok(id)
```

Comment:
why `self.indexed` is no more updated here ?

Reply:
this method is called whenever a new field is found. I want to add it to the known fields (`field_map`), and generate an indexed position for it, by inserting it in `indexed_map`. if I added it to  `indexed` too, then it would be considered an indexed field and this is what we do not want. I guess we could rename this method `add_known_field`, what do you think?

Reply:
So what's the purpose of `indexed_map`? Why this name if this is not here to store the indexed fields?

Reply:
Yeah, we can maybe find a better name than `insert_and_index`, indeed.

Reply:
I don't understand everything.

For a user we have 3 states for an attribute:
`indexed`: indexed in meilisearch but not used 
`displayed`: displayed in a search result
`searchable`: used when I search something

Internally their are:
`fields_map`
`indexed_map`
`indexed`
`displayed_map`

what is what ? 🤔